main.js

```diff
@@ -0,0 +1,147 @@
+// Get the HTML element by ID for search button
+const searchButton = document.getElementById('search-button')
+
+// Event listener for the searchbutton on click
+searchButton.addEventListener('click', fetchData)
+
+// Get the HTML element by ID for the form
+const form = document.querySelector('#my-form')
+
+// Create a function to submit the users input and fetch the data from the fetchData function
+form.onsubmit = function (e) {
+    e.preventDefault();
+    return fetchData()
+}
+
+// Get the HTML element by ID for the image
+const flag = document.getElementById('flag')
+
+// Get the HTML element by ID for the country name
+const countryName = document.getElementById('country-name')
+
+// Get the HTML element by ID for the info
+const info = document.getElementById('info')
+
+// Get the HTML element by ID for the capital
+const capital = document.getElementById('capital')
+
+// Get the HTML element by ID for the language
+const languageInfo = document.getElementById('language')
+
+let searchBar = document.getElementById('search-bar')
+function searchValue() {
+    return searchBar.value
+}
+
+// Create a function to clear the form after user input
+function resetForm() {
+    return document.getElementById('search-bar').value = ''
+}
+
+// Create a function to clear the error message on a new user input
+function resetErrorField() {
+    const message = document.getElementById('error-message')
+    message.innerHTML = ''
+}
+
+// Function to get the data from the uri
+async function fetchData() {
+    try {
+        let search = searchValue()
+        let country = await axios.get(`https://restcountries.eu/rest/v2/name/${search}`)
```

Comment:
Wat ik zelf heel vervelend vind is dat ik heel veel `country.data[0]` moet herhalen. Wij developers zijn heel lui, dus is daarom de object spreader in het leven geroepen 😉 
```
const { name: countryName, population, subregion: sbuAreaName, ...otherProps } = country.data[0];
```

_Je hoeft de props niet perse te aliasen_